src/components/20-molecules/datepicker/index.js

```diff
@@ -332,24 +352,8 @@ class AXADatepicker extends NoShadowDOM {
     this.startDate.setHours(0);
     this.startDate.setMinutes(0);
     this.startDate.setSeconds(0);
-    // the chosen start year is in the allowed years
-    let allowedYearsFinal = [this.year];
-    this.allowedyears.forEach(years => {
-      if (typeof years === 'string') {
-        const splitYears = years.split('-');
-        const generatedYears = range(
-          parseInt(splitYears[0], 10),
-          parseInt(splitYears[1], 10)
-        );
-        allowedYearsFinal = allowedYearsFinal.concat(generatedYears);
-      } else {
-        allowedYearsFinal.push(years);
-      }
-    });
 
-    this.allowedyears = allowedYearsFinal
-      .filter((item, index) => allowedYearsFinal.indexOf(item) >= index)
-      .sort();
+    this.allowedyears = parseAndFormatAllowedYears(this);
```

Comment:
Maybe more readable, if you can only pass stuff thats needed in the function. Then the function have no reason to destruct a object.